### PR TITLE
very boring PR

### PR DESCRIPTION
**Pull request by GrahamSH-LLK**
run it and find out??
currently bound to <kbd>s</kbd>

### CODE REVIEW

File: addons/fools/index.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  });
+}
+const wait = (s) =>
+  new Promise((resolve, reject) => {
+    setTimeout(() => {
+      return resolve();
+    }, s);
+  });
+async function setup() {
+  document.querySelector("#view > div > div:nth-child(3) > div:nth-child(1)").scrollIntoView();
```
Comment (Samq64) [inline addons/fools/index.js:15]: Uncaught (in promise) TypeError: Cannot read properties of null (reading 'scrollIntoView')

Comment (GrahamSH-LLK) [reply addons/fools/index.js:15]: lmao well that's interesting

Comment (GrahamSH-LLK) [reply addons/fools/index.js:15]: works on my device, lemme figure out why

Comment (GrahamSH-LLK) [reply addons/fools/index.js:15]: for now test while logged out, i have some dumb selectors

**Comment by Samq64** [comment]
Could the it be contained within `mod-splash`?

File: addons/fools/index.js (hunk 1 above)
Comment (Waakul) [reply addons/fools/index.js:15]: > Uncaught (in promise) TypeError: Cannot read properties of null (reading 'scrollIntoView')

got the same

**Comment by OneShot-Niko** [comment]
Spamming or pressing S again after starting lags it a lot - This is small but I think I should point that out.

File: addons/fools/addon.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "eastah egg",
+  "description": "o_0",
+  "tags": ["community"],
```
Comment (OneShot-Niko) [inline addons/fools/addon.json:4]: This is probably not neccasary to do at all, but
```suggestion
  "tags": ["community", "new"],
```

**Comment by Waakul** [comment]
why dont make prettier happy?

File: addons/fools/addon.json (hunk 2 above)
Comment (Samq64) [reply addons/fools/addon.json:4]: The new tag is automatically added based on `versionAdded` and [latestUpdate](https://scratchaddons.com/docs/reference/addon-manifest/#latestupdate).

Comment (OneShot-Niko) [reply addons/fools/addon.json:4]: > The new tag is automatically added based on `versionAdded` and [latestUpdate](https://scratchaddons.com/docs/reference/addon-manifest/#latestupdate).

Ah, my bad then.

**Comment by GrahamSH-LLK** [comment]
> why dont make prettier happy?

prettier is happy, the action is broken

**Comment by GrahamSH-LLK** [comment]
> This also triggers when typing into a text box. @Samq64 
> Spamming or pressing S again after starting lags it a lot - This is small but I think I should point that out. (EDIT: Considering it does restart other things such as the ball's position)

there will be a button hidden somewhere, S is for testing

**Comment by OneShot-Niko** [comment]
> > This also triggers when typing into a text box. @Samq64 
> > Spamming or pressing S again after starting lags it a lot - This is small but I think I should point that out. (EDIT: Considering it does restart other things such as the ball's position)
> 
> there will be a button hidden somewhere, S is for testing

My bad then.

File: addons/addons.json [hunk 3]
```diff
@@ -153,7 +153,8 @@
   "collapse-footer",
   "reorder-custom-inputs",
   "place-backpack-code-at-cursor",
-
+  "fools",
```
Comment (Samq64) [inline addons/addons.json:156]:
```suggestion
  "breakout",
```

File: addons/fools/addon.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "eastah egg",
```
Comment (Samq64) [inline addons/fools/addon.json:2]:
```suggestion
  "name": "Breakout",
```

File: addons/fools/addon.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  "description": "o_0",
```
Comment (Samq64) [inline addons/fools/addon.json:3]:
```suggestion
  "description": "Allows you to play Breakout with the thumbnails on Scratch's home page.",
```

File: addons/fools/addon.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  "tags": ["community"],
+  "enabledByDefault": false,
```
Comment (Samq64) [inline addons/fools/addon.json:5]: We'll probably end up doing some hardcoded logic for April Fools day like other years, right?

File: addons/fools/index.js [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    if (e.key == "s") {
+      await setup();
+    }
+  });
+  async function setup() {
+    await wait(300);
+    document.body.style.overflow = "hidden";
+    // create full screen canvas
+    const canvas = document.createElement("canvas");
+    canvas.width = window.innerWidth;
```
Comment (Samq64) [inline addons/fools/index.js:12]: The page's content doesn't get wider than 942px which means no thumbnails so I think the canvas should be limited to around there, maybe with the score/lives still in the corners?
```suggestion
    canvas.width = 950;
```

File: addons/fools/index.js [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+    canvas.height = window.innerHeight;
+    canvas.style.position = "fixed";
+    canvas.style.top = "0";
+    canvas.style.left = "0";
+    canvas.style.zIndex = "1000";
+    document.body.appendChild(canvas);
+    canvas.style.transition = "opacity 3s ease-in-out";
+    canvas.style.opacity = "0.1";
+
+    canvas.style.backgroundColor = "black";
```
Comment (Samq64) [inline addons/fools/index.js:22]: Just going on the homepage now there are some thumbnails with pure black backgrounds.
```suggestion
    canvas.style.backgroundColor = "#111";
```

File: addons/fools/index.js [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+      ctx.beginPath();
+      ctx.arc(x, y, ballRadius, 0, Math.PI * 2);
+      ctx.fillStyle = "#0095DD";
+      ctx.fill();
+      ctx.closePath();
+    }
+
+    function drawPaddle() {
+      ctx.beginPath();
+      ctx.rect(paddleX, canvas.height - paddleHeight, paddleWidth, paddleHeight);
```
Comment (Samq64) [inline addons/fools/index.js:113]: Could the paddle be raised off the bottom of the window a little?

File: addons/fools/index.js [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+
+      x += dx;
+      y += dy;
+      if (gameOver) {
+        canvas.remove();
+        document.body.style.overflow = "auto";
+        if (score === brickRowCount * brickColumnCount * 3) {
+          addon.tab.confirm("You won!", "Happy april fools from the Scratch Addons team!");
+        } else {
+          addon.tab.confirm("Game over!", "Happy april fools from the Scratch Addons team!");
```
Comment (Samq64) [inline addons/fools/index.js:218]: These will need to be translatable.

**Comment by Samq64** [comment]
Button position suggestions:

**Review by Joeclinton1 — CHANGES_REQUESTED** [review]
I agree with @Samq64 's changes, plus I have my own set of changes. Here's what I'd like changed:

- you've just hid the background but kept the images in place. This isn't enough. You need to do the more complex task of resizing the images into block shaped images and moving them to the top of the screen. You'll need more than just 15 blocks for this though. This makes the block thin and removes gaps between blocks. This would greatly enhance the game.
- Like the google one, it would be nice to have the colour overlays so that it looks more like breakout.
- the ball needs to move about 1.5x as fast or there's no challenge
- The blocks need to take up the space up to the edges of the playable area, this is crucial to the game being fun.
- The collisions on the ball are quite wrong. It must collide with the boxes at the actual ball edge point.
- It of course needs a way to start the game, beyond clicking s.
- The score and lives design could be improved
- it would be nice for it to also work on the explore page. Which i believe shouldn't be too much work if you can get it to work on the homepage. You could even make your code flexible to work on any page with thumbnails.
- Be nice if you could add the sound effects too

This currently is about 50% of the way there though, so well done for that. The closer you can get to https://elgoog.im/breakout/ the better.

File: addons/breakout/addon.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Breakout",
+  "description": "Allows you to play Breakout with the thumbnails on Scratch's home page.",
+  "tags": ["community"],
```
Comment (Samq64) [inline addons/breakout/addon.json:4]:
```suggestion
  "tags": ["community", easterEgg],
```

**Review by Samq64 — COMMENTED** [review]
For reducing the width I forgot to mention the rest of the page will still need to be covered by something.

File: addons/breakout/addon.json (hunk 11 above)
Comment (Samq64) [reply addons/breakout/addon.json:4]: Oops.
```suggestion
  "tags": ["community", "easterEgg"],
```

**Comment by mybearworld** [comment]
I was about to say that it's really early for April Fools preparations and then noticed it's really only a month away. Huh.

**Comment by GrahamSH-LLK** [comment]
> I was about to say that it's really early for April Fools preparations and then noticed it's really only a month away. Huh.

wdym no april fools here you're crazy

**Comment by Waakul** [comment]
Happy april fools popup should only appear if it's april fools

**Comment by Waakul** [comment]
The score text and lives text very hard to see

**Comment by mybearworld** [comment]
> Happy april fools popup should only appear if it's april fools

This is an addon specifically for April Fool's.

**Comment by Waakul** [comment]
enable this by default?

**Comment by Waakul** [comment]
> enable this by default?

also this addon should only be visible on april fools, we'll need to create some kind of tag for that

**Comment by BroJac5246** [comment]
> > enable this by default?
> 
> also this addon should only be visible on april fools, we'll need to create some kind of tag for that

I believe we just manually check the date. However, because that makes it more difficult to test, that hasn't been added yet. Once it's ready, the PR will be marked as ready for review and then details such as the ones you mentioned will be looked over.

**Comment by Waakul** [comment]
> > > enable this by default?
> > 
> > 
> > also this addon should only be visible on april fools, we'll need to create some kind of tag for that
> 
> I believe we just manually check the date. However, because that makes it more difficult to test, that hasn't been added yet. Once it's ready, the PR will be marked as ready for review and then details such as the ones you mentioned will be looked over.

huh? there will be background page logics that will check if it's april fools to run this (this wont be an addon).

**Comment by Waakul** [comment]
> > > enable this by default?
> > 
> > 
> > also this addon should only be visible on april fools, we'll need to create some kind of tag for that
> 
> I believe we just manually check the date. However, because that makes it more difficult to test, that hasn't been added yet. Once it's ready, the PR will be marked as ready for review and then details such as the ones you mentioned will be looked over.

it's best to have a tag and some bg logic that does this

**Comment by Waakul** [comment]
> > > > enable this by default?
> > > 
> > > 
> > > also this addon should only be visible on april fools, we'll need to create some kind of tag for that
> > 
> > 
> > I believe we just manually check the date. However, because that makes it more difficult to test, that hasn't been added yet. Once it's ready, the PR will be marked as ready for review and then details such as the ones you mentioned will be looked over.
> 
> huh? there will be background page logics that will check if it's april fools to run this, it's best to add it in a tag

**Comment by Samq64** [comment]
> huh? there will be background page logics that will check if it's april fools to run this

[Here's](https://github.com/ScratchAddons/ScratchAddons/pull/5831/files#diff-716aae76bfb4eaa095810abaddc214bb23abc4a9ddc232f27cb3203233c1c0da) last year's logic.

**Comment by Waakul** [comment]
> > huh? there will be background page logics that will check if it's april fools to run this
> 
> [Here's](https://github.com/ScratchAddons/ScratchAddons/pull/5831/files#diff-716aae76bfb4eaa095810abaddc214bb23abc4a9ddc232f27cb3203233c1c0da) last year's logic.

i dont think it enables it by default if it's april fools

**Comment by WorldLanguages** [comment]
Please don't go off-topic. Adding the logic so that this only works on April Fools is the simplest part.

**Comment by Samq64** [comment]
@GrahamSH-LLK We're running out of time. Also when I suggested shrinking the canvas I didn't think of the rest of the page. There should probably be a full-size overlay element between the page and the canvas.

**Comment by mybearworld** [comment]
> there will be a button hidden somewhere, S is for testing

So... where should that button be?

**Comment by WorldLanguages** [comment]
I think we're going with #7299 instead.

**Comment by Waakul** [comment]
> > there will be a button hidden somewhere, S is for testing
> 
> So... where should that button be?

That button is already added, In the nav bar near the messages button.

**Comment by Samq64** [comment]
I think this would still be a cool easter egg to have around.

**Comment by BroJac5246** [comment]
> I think this would still be a cool easter egg to have around.

We chould still try to use it next year.

**Comment by BlueGoat11** [comment]
A cool Easter egg would be like a Konami code to unlock something or something like that…

**Comment by BroJac5246** [comment]
> A cool Easter egg would be like a Konami code to unlock something or something like that…

For example a whole collection of easter egg addons? #1423

**Comment by BlueGoat11** [comment]
Something like that… Another idea is maybe replacing the background of the code area with an April fools collage. I’m just spitting out ideas at this point though.

**Pull request closed without merging**